File: schrottverwaltung/src/main/java/de/fhdortmund/schrottverwaltung/bewertung/Bewertung.java
package de.fhdortmund.schrottverwaltung.bewertung;

import de.fhdortmund.schrottverwaltung.immobilie.Immobilie;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "bewertung")
@Getter
@Setter
public class Bewertung {

    @Column(name = "gewichtung")
    private int gewichtung;

    @ManyToOne
    @JoinColumn(name = "kriterium_id")
    private Kriterium kriterium;

    @ManyToOne
    @JoinColumn(name = "immobilie_id") // <- Referenz zur Immobilie
    private Immobilie immobilie;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "bewertung_id")
    private Long id;
}

File: schrottverwaltung/src/main/kotlin/de/fhdortmund/schrottverwaltung/bewertung/Bewertung.kt
package de.fhdortmund.schrottverwaltung.bewertung

import de.fhdortmund.schrottverwaltung.immoblilie.Immobilie
import jakarta.persistence.*

@Entity
@Table(name = "bewertung")
data class Bewertung(

    @Column(name = "gewichtung")
    var gewichtung: Int,
    @ManyToOne
    @JoinColumn(name = "kriterium_id")
    var kriterium: Kriterium,
    @ManyToOne
    @JoinColumn(name = "immobilie_id") // <- Referenz zur Immobilie
    var immobilie: Immobilie,
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "bewertung_id")
    var id: Long? = null
)
